test/csp_helpers.py

```diff
@@ -376,6 +380,25 @@ def __init__(self):
         )
         self.metadata_token = "test-token"
 
+    def assume_role(self, **kwargs):
+        if (
+            self.assumption_path
+            and kwargs["RoleArn"] == self.assumption_path[self.assume_role_call_count]
+        ):
+            arn = self.assumption_path[self.assume_role_call_count]
+            self.assume_role_call_count += 1
+            return {
+                "Credentials": {
+                    "AccessKeyId": gen_dummy_access_token(arn, "access_key"),
+                    "SecretAccessKey": gen_dummy_access_token(arn, "secret_key"),
+                    "SessionToken": gen_dummy_access_token(arn, "session_token"),
```

Comment:
tiny nit: those don't need to be "tokens" by the typical definition of the word. You can just use random string placeholders here to simplify and avoid confusion:

```suggestion
                    "AccessKeyId": "access_key_id",
                    "SecretAccessKey": "secret_access_key",
                    "SessionToken": "session_token",
```